python/teams_ai/turn_state/turn_state_entry.py

```diff
@@ -3,58 +3,66 @@
 Licensed under the MIT License.
 """
 
-from typing import TypeVar, Generic, TypedDict, Union
+from typing import TypeVar, Generic, Optional
 
-T = TypeVar("T", TypedDict)
+ValueT = TypeVar("ValueT")
 
 
-class TurnStateEntry(Generic[T]):
+class TurnStateEntry(Generic[ValueT]):
     "accessor class for managing an individual state scope"
 
-    __value: T
-    __storage_key: Union[str, None]
-    __deleted = False
-    __hash: str
+    _value: ValueT
+    _storage_key: Optional[str]
+    _hash: str
+    _deleted = False
 
-    def __init__(self, value: T, storage_key=None) -> None:
-        self.__value = value
-        self.__storage_key = storage_key
-        self.__hash = value.__str__()
+    def __init__(self,
+                 value: Optional[ValueT] = None,
+                 storage_key: Optional[str] = None) -> None:
+        self._value = value
+        self._storage_key = storage_key
+        self._hash = str(self._value)
 
     @property
     def changed(self) -> bool:
         "gets a value indicating whether the state scope has changed since it was last loaded"
 
-        return self.__hash != str(self.__value)
+        return self._hash != str(self._value)
 
     @property
     def deleted(self) -> bool:
         "gets the value indicating whether the state scope has been deleted"
 
-        return self.__deleted
+        return self._deleted
 
     @property
-    def value(self) -> T:
-        "gets the value of the state scope"
-
-        if self.__deleted:
-            self.__value = {}
-            self.__deleted = False
+    def empty(self) -> bool:
+        "is the stored values empty"
 
-        return self.__value
+        return self._value is None
 
     @property
-    def storage_key(self) -> Union[str, None]:
+    def storage_key(self) -> Optional[str]:
         "gets the storage key used to persist the state scope"
 
-        return self.__storage_key
+        return self._storage_key
+
+    @property
+    def value(self) -> ValueT:
+        "gets the value of the state scope"
+
+        if self.deleted:
+            self._deleted = False
+
+        return self._value
 
-    def delete(self) -> None:
-        "clears the state scope"
+    @value.setter
+    def value(self, value: ValueT):
```

Comment:
should we use `replace()` function instead of a property setter here to align with other SDK?